SpiceSharp/Simulations/IntegrationMethods/Spice/Trapezoidal/Trapezoidal.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using SpiceSharp.Simulations;
+
+namespace SpiceSharp.IntegrationMethods
+{
+    /// <summary>
+    /// Trapezoidal integration method
+    /// </summary>
+    public partial class Trapezoidal : SpiceIntegrationMethod
+    {
+        /// <summary>
+        /// Integration coefficients
+        /// </summary>
+        protected double[] Coefficients { get; } = new double[2];
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public Trapezoidal()
+            : base(2)
+        {
+        }
+
+        /// <summary>
+        /// Initialize the trapezoidal integration method
+        /// </summary>
+        /// <param name="simulation">The simulation</param>
+        public override void Initialize(TimeSimulation simulation)
+        {
+            base.Initialize(simulation);
+
+            // Reset all coefficients
+            for (var i = 0; i < MaxOrder; i++)
+                Coefficients[i] = 0.0;
+        }
+
+        /// <summary>
+        /// Unsetup the integration method
+        /// </summary>
+        public override void Unsetup()
+        {
+            base.Unsetup();
+
+            // Reset all coefficients
+            for (var i = 0; i < MaxOrder; i++)
+                Coefficients[i] = 0.0;
+        }
+
+        /// <summary>
+        /// Predict a new solution based on the previous ones
+        /// </summary>
+        /// <param name="simulation">Time-based simulation</param>
+        protected override void Predict(TimeSimulation simulation)
+        {
+            if (simulation == null)
+                throw new ArgumentNullException(nameof(simulation));
+
+            // Use the two previous solutions to predict a new one (the one we're about to test)
+            var future = IntegrationStates[0];
```

Comment:
@svenboulanger  I like it :) Now I now what is IntegrationStates[0] and etc.

Reply:
This is the method for predicting a solution. In my mind we *predict* a future solution, but it is actually predicting the solution we're now trying to calculate, ie. at the current time point.

So it's true that the index 0 in this case was called future rather than current. Can be interpreted both ways. Should not affect behavior too much.